Src/BootCamp.Chapter/ApplicationController.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Reflection;
+
+namespace BootCamp.Chapter
+{
+    class ApplicationController
+    {
+        // implement into a method that invokes this event and has the termination logic tirggered?
+        public event EventHandler<ApplicationModeEventArgs> ApplicationModeChanged;
```

Comment:
Instead of having one event with arguments for each state, have multiple events, without any arguments. That way, you can freely subscribe only to the state of application that you are interested in. Also, you won't have to do an if statement to check what state it is in. Do something when it Start, when it Ends. Not always.